### PR TITLE
prevent db connections during asset precompile

### PR DESCRIPTION
**Pull request by karlhungus**
#### Problem

We are creating db connections during asset compilation

#### Solution

Do connections on load

#### Problem

We are creating db connections during asset compilation

#### Solution

Do connections on load

Full disclosure: i have no idea what i'm doing :(

@kirs @rafaelfranca /cc @celsodantas

### CODE REVIEW

File: lib/bugsnag/railtie.rb [hunk 1]
```diff
@@ -45,8 +45,10 @@ class Railtie < Rails::Railtie
         ActionController::API.send(:include, Bugsnag::Rails::ControllerMethods)
       end
       if defined?(ActiveRecord::Base)
-        require "bugsnag/rails/active_record_rescue"
-        ActiveRecord::Base.send(:include, Bugsnag::Rails::ActiveRecordRescue)
+        ActiveSupport.on_load(:active_record) do
+          require "bugsnag/rails/active_record_rescue"
+          ActiveRecord::Base.send(:include, Bugsnag::Rails::ActiveRecordRescue)
```
Comment (rafaelfranca) [inline lib/bugsnag/railtie.rb]: `include Bugsnag::Rails::ActiveRecordRescue`

File: lib/bugsnag/railtie.rb [hunk 2]
```diff
@@ -45,8 +45,10 @@ class Railtie < Rails::Railtie
         ActionController::API.send(:include, Bugsnag::Rails::ControllerMethods)
       end
       if defined?(ActiveRecord::Base)
```
Comment (rafaelfranca) [inline lib/bugsnag/railtie.rb]: This is not needed anymore.

**Comment by rafaelfranca** [comment]
This is also related with https://github.com/bugsnag/bugsnag-ruby/pull/314

**Comment by kattrali** [comment]
This makes sense. :) Thank you for the contribution, @karlhungus, and for looking it over, @rafaelfranca.

**Pull request merged**